.github/workflows/codeql-analysis.yml

```diff
@@ -79,3 +79,19 @@ jobs:
 
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@b398f525a5587552e573b247ac661067fafa920b # tag=v2.1.22
+
+  # NOTE: Checks that the matrix job above completes successfully.
+  # This is necessary because the matrix strategy generates new jobs with
+  # different names and throws off the required checks on protected
+  # branches which are set by job name.
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    needs: [analyze-matrix]
+    if: ${{ always() }}
+    env:
+      ANALYZE_RESULT: ${{ needs.analyze-matrix.result }}
```

Comment:
so this will be "success" only if all matrix jobs succeed, correct? If so LGTM.

Reply:
correct